.github/workflows/release.yml

```diff
@@ -0,0 +1,81 @@
+# This workflow is mostly copied from the official PyPA guide:
+# https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+#
+# Note: There are improvements that could be made to streamline the release process further such as integrating helpers
+#       like bump-my-version and towncrier
+#
+# Instructions:
+#
+# 1. Finalise development on the master branch
+# 2. Update the version number in setup.py appropriately
+# 3. Commit `git commit -m "vx.x.x release"`
+# 4. Tag `git tag vx.x.x`
+# 5. Push `git push && git push --tags`
+# 6. This workflow will automatically be triggered in GitHub Actions based on the new tag
+# 7. Update the version number in setup.py again to a development version e.g. v1.2.3.dev0
```

Comment:
This would be in pyproject.toml now 😅 
I will take a look for an action to automate this, I can see that being the step that gets forgotton..